Add barracuda/0.1.0 integration (snapshot)

adriansr opened the pull request:
This adds the first rsa2elk integration package for Barracuda WAF devices.

After this one, I'll add the next 19 rsa2elk devices in a single PR if possible. They all follow the exact same structure than this one.


## Screenshots

<img width="1250" alt="Screenshot 2020-07-23 at 15 41 58" src="https://user-images.githubusercontent.com/15056957/88293428-484bb480-ccfb-11ea-8277-dd02d8f1d769.png">

<img width="433" alt="Screenshot 2020-07-23 at 15 42 38" src="https://user-images.githubusercontent.com/15056957/88293439-4c77d200-ccfb-11ea-8c09-a2e0528f8c14.png">


With logo:

<img width="678" alt="Screenshot 2020-07-28 at 12 02 13" src="https://user-images.githubusercontent.com/15056957/88651767-510d0380-d0ca-11ea-9114-1efc3bc6cc9c.png">

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackage-storage%2FPR-152/detail/PR-152/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackage-storage%2FPR-152/detail/PR-152/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackage-storage%2FPR-152/detail/PR-152/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fpackage-storage%2FPR-152/detail/PR-152/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://package-storage_152.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #152 opened]`

* Start Time: 2020-07-22T16:57:47.536+0000

* Duration: 8 min 17 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 0  |
| Passed       | 24  |
| Skipped      | 0 |
| Total        | 24   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->





<!--COMMENT_GENERATED_WITH_ID_comment.id-->

mtojek commented on packages/barracuda/0.1.0/dataset/waf/agent/stream/stream.yml.hbs:
```diff
@@ -0,0 +1,3721 @@
+      keep_raw: {{keep_raw_fields}}
+      debug: {{debug}}
+    source: |
+      // Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
@nchaulet @michalpristas Could you please confirm that such huge blobs should be supported by all parties?

mtojek commented on packages/barracuda/0.1.0/dataset/waf/manifest.yml:
```diff
@@ -0,0 +1,155 @@
+    default: localhost
+  - name: tcp_port
+    type: integer
+    title: TCP Port to listen on
```
nit: port

andrewstucki commented on packages/barracuda/0.1.0/manifest.yml:
```diff
@@ -0,0 +1,31 @@
+        title: Collect logs from Barracuda Web Application Firewall via file
+        description: Collecting syslog from Barracuda Web Application Firewall via file.
+# No icon
+icon:
```
I'm pretty sure we can easily find a barracuda svg to dump on here, is there a reason not to add an icon?

adriansr replied:
Afaik we don't have (yet) clearance from Barracuda for the use of their logo in a commercial product. I see a lot of logos in other packages, @mtojek is someone from the ingest manager team handling this? /cc @jamiehynds

mtojek replied:
Nobody. Currently used logos are imported from existing open-source repos of Elastic. In general in terms of the ingest manager, the point of contact is @ruflin .

andrewstucki replied:
So, I pulled the logo from https://www.elastic.co/integrations for my WIP integration. Even though I once went to law school, IANAL, so not entirely sure what additional consent we need from those other companies for inclusion of their logos here.

adriansr replied:
According to @jamiehynds we're OK adding the logos. Will update the PR.

andrewstucki commented on packages/barracuda/0.1.0/docs/README.md:
```diff
@@ -0,0 +1,9 @@
+
+### Waf
+
+The `waf` dataset collects Barracuda Web Application Firewall logs.
```
So, generally the way this file gets generated in the integrations directory, it contains a listing of all of the fields along with their documentation. Does this need to be consistent here too?

andrewstucki commented on packages/barracuda/0.1.0/dataset/waf/agent/stream/udp.yml.hbs:
```diff
@@ -0,0 +1,3716 @@
+udp:
+host: "{{udp_host}}:{{udp_port}}"
+tags: {{tags}}
```
See [comment](https://github.com/elastic/integrations/pull/186#discussion_r458441240) I made on the Suricata package, but in filebeat we pipe this to `toJSON`, so I'm assuming we need it rendered as a yaml array. The way this is right now it'll output as a comma-separated string.

adriansr commented:
- Changed categorization to Network + Security
- Added logo:

<img width="678" alt="Screenshot 2020-07-28 at 12 02 13" src="https://user-images.githubusercontent.com/15056957/88651865-6bdf7800-d0ca-11ea-8ba4-9a3426aee316.png">
